atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceImpl.java

```diff
@@ -2069,6 +2072,67 @@ public void deleteRange(final TableReference tableRef, final RangeRequest range)
         }
     }
 
+    @Override
+    public void deleteAllTimestamps(TableReference tableRef, Map<Cell, Long> maxTimestampExclusiveByCell) {
+        Map<InetSocketAddress, Map<Cell, Long>> keysByHost = partitionMapByHost(maxTimestampExclusiveByCell.entrySet());
+        for (Map.Entry<InetSocketAddress, Map<Cell, Long>> entry : keysByHost.entrySet()) {
+            deleteAllTimestampsOnSingleHost(tableRef, entry.getKey(), entry.getValue());
+        }
+    }
+
+    public void deleteAllTimestampsOnSingleHost(
+            TableReference tableRef,
+            InetSocketAddress host,
+            Map<Cell, Long> maxTimestampExclusiveByCell) {
+        if (maxTimestampExclusiveByCell.isEmpty()) {
+            return;
+        }
+
+        try {
+            clientPool.runWithRetryOnHost(host, new FunctionCheckedException<CassandraClient, Void, Exception>() {
+
+                @Override
+                public Void apply(CassandraClient client) throws Exception {
+                    Map<ByteBuffer, Map<String, List<Mutation>>> mutationsByRowByTable = Maps.newHashMap();
+
+                    maxTimestampExclusiveByCell.forEach((cell, maxTimestampExclusive) -> {
+                        SlicePredicate pred = SlicePredicates.rangeTombstoneForColumn(
+                                cell.getColumnName(), maxTimestampExclusive);
+
+                        Deletion del = new Deletion();
+                        del.setPredicate(pred);
+                        // TODO(nziebart): figure out if/why we can't use the provided max timestamp for the cell here
+                        del.setTimestamp(Long.MAX_VALUE);
+
+                        Mutation mutation = new Mutation();
```

Comment:
Thought: I'm wondering if we want helper methods that would take a `columnName` and `maxTimestampExclusive`, and give us a "delete range tombstone" mutation.

This would help make this method more readable by encapsulating the boilerplate like "ok, I want to have a Mutation, but first I need a Deletion, and oh jeez, that means I want this slice predicate and I have to hard-code `Long.MAX_VALUE`" - instead, the method becomes "cool, I pass the column name and max timestamp into this helper method and add it to my collection".

Reply:
done